File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:


        result = [0] * len(nums)

        left = 0
        right = len(nums) - 1
        k = len(nums) - 1

        while left <= right:

            a = abs(nums[left])
            b = abs(nums[right])

            if a > b:
                result[k] = a * a
                left += 1
            else:
                result[k] = b * b
                right -= 1

            k -= 1

        return result
        

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function (nums) {

    let result = new Array(nums.length);

    let left = 0
    let right = nums.length - 1;
    let k = nums.length - 1;

    while (left <= right) {
        let a = Math.abs(nums[left]);
        let b = Math.abs(nums[right]);

        if (a > b) {
            result[k] = a * a;
            left++;
        
        } else {

            result[k] = b * b;
            right--;
        }
        k--;
    }
    return result;






















// Brute Force

// for(let i=0; i<nums.length; i++){
//     nums[i]= nums[i]*nums[i];
// }

// nums.sort((a,b) => a-b);

// return nums;
    
};